### PR TITLE
Feature/channels

### PR DESCRIPTION
**Pull request by mpowaga**

### CODE REVIEW

**Comment by ricricucit** [comment]
Thanks for this, @mpowaga.
I can see that some tests are failing, please ensure that they are passing.
Some of these are unrelated to your work (related to CLI), but 2 of these are:

1. Channel can reestablish a channel – can reestablish a channel
```
Stacktrace
Error: Timeout of 180000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/app/test/integration/channel.js)
```
2. Channel can leave a channel – can leave a channel
```
Stacktrace
Error: Timeout of 180000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/app/test/integration/channel.js)
```

You can check [here](https://ci.aepps.com/blue/organizations/jenkins/aepp-sdk-js/detail/feature%2Fchannels/12/tests) the errors I'm referring to.

Thanks,
Enrico.

**Comment by mpowaga** [comment]
@ricricucit There's a problem that sometimes epoch channel gets stuck and doesn't move to next state. This is often when opening a channel or sending an update.

I've fixed following error which was caused by a bug in implementation:

`Error: State Channels FSM entered unknown state`

But now test failed to open a channel. It happens quite often when I'm testing on my local machine. Usually this can be fixed by simply rerunning tests.

We can try to implement reconnect strategy but I'm not sure if it's worth if it's epoch bug.

**Comment by ricricucit** [comment]
Now it's passing. @mpowaga can you do some more testing to see if that issue got fixed in [this release](https://github.com/aeternity/epoch/releases/tag/v0.25.0)? Thanks!

**Comment by ricricucit** [comment]
@mpowaga could be this considered done? Is there anything to add?

**Comment by mpowaga** [comment]
@ricricucit I think that cli tool needs to be separated now.

**Comment by ricricucit** [comment]
Cool. 
Can you please push the `aecli` files to https://github.com/aeternity/aecli-js and leave only SDK related files in this PR? Thanks!

**Comment by mpowaga** [comment]
@ricricucit done.

File: docker-compose.yml [hunk 1]
```diff
@@ -27,6 +27,14 @@ services:
       - node_db:/home/epoch/node/data/mnesia
       - node_keys:/home/epoch/node/keys
 
+  proxy:
```
Comment (ricricucit) [inline docker-compose.yml]: Do we actually need this?
As of [0.24.0](https://github.com/aeternity/epoch/blob/v0.24.0/docs/release-notes/RELEASE-NOTES-0.24.0.md) CORS are included in Epoch.

File: docker/nginx-cors.conf [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#
```
Comment (ricricucit) [inline docker/nginx-cors.conf]: _Same as before (and for all the `.conf` files)_ 😅

Do we actually need this?
As of 0.24.0 CORS are included in Epoch.

**Comment by mpowaga** [comment]
@ricricucit I've removed proxy.

**Comment by ricricucit** [comment]
@mpowaga for some reason, tests always get stuck and stay there forever...and I have to stop them manually. E.g.: https://ci.aepps.com/blue/organizations/jenkins/aepp-sdk-js/detail/PR-93/27/pipeline

**Comment by mpowaga** [comment]
@ricricucit Following tests don't work with epoch `v1.0.0`.

- can post update and reject
- can close a channel
- can reestablish a channel

Perhaps `mocha.retries(3)` should be reduced to `mocha.retries(1)` or removed.

**Comment by ricricucit** [comment]
Mmmmhh...can you check this issue?

**Comment by mpowaga** [comment]
@ricricucit These issues need to be fixed in epoch so we need to wait a bit longer.

**Comment by ricricucit** [comment]
Is anyone in the core team aware of this?

**Comment by mpowaga** [comment]
Yes, I talked with Dimitar, though we could also let them know that this PR is blocked by these issues.

**Comment by ricricucit** [comment]
Cool, thanks. Ok, so let's see what happens if I mention @velzevur here! 😊

**Comment by ricricucit** [comment]
Glad to see tests passing! 👏

**Pull request merged**